### PR TITLE
chore(dependencies): update tokio, h2, and tower-make

### LINKED ISSUE #1945: Update to tokio alpha 5
- [ ] update h2
- [ ] update hyper

### PR DESCRIPTION
**Pull request by taiki-e**
Closes #1945

### CODE REVIEW

**Comment by taiki-e** [comment]
The cause of test failure seems to be https://github.com/rust-lang/rust/issues/64636.

**Comment by seanmonstar** [comment]
Thanks for this! I got the permissions to publish tokio-make, so published alpha2, so we can update here!

File: examples/send_file.rs [hunk 1]
```diff
@@ -61,6 +61,7 @@ fn internal_server_error() -> Response<Body> {
         .unwrap()
 }
 
+#[allow(unreachable_code)] // TODO: remove this when https://github.com/rust-lang/rust/issues/64636 fixed.
```
Comment (taiki-e) [inline examples/send_file.rs]: Also, as https://github.com/rust-lang/rust/issues/64636 has been fixed, we can remove this.

**Comment by taiki-e** [comment]
@seanmonstar Updated and CI passed.

**Comment by seanmonstar** [comment]
Awesome, thanks!

**Pull request merged**